### PR TITLE
Represent integer literals as `IntLiteral` not as `i32`.

### PR DESCRIPTION
**Pull request by zygoloid**
When an `IntLiteral` appears as an operand of an `if` expression, convert it to `i32` for now, so that we don't reject things like `if cond then 1 else 2` due to having a non-constant value of type `IntLiteral`.

For tuple indexing expressions such as `(a, b).0`, convert the index to type `IntLiteral`, not to type `i32`. This isn't strictly necessary to do in this PR, but avoids the need to provide an `IntLiteral` -> `i32` implicit conversion for `no_prelude` tests using this syntax.

### CODE REVIEW

**Comment by zygoloid** [comment]
Depends on #4526.

File: core/prelude/operators/as.carbon [hunk 1]
```diff
@@ -12,3 +16,54 @@ interface ImplicitAs(Dest:! type) {
   // TODO: extend As(Dest);
   fn Convert[self: Self]() -> Dest;
 }
+
+// TODO: These impls should live with Core.Int, but currently that's a builtin
+// not a class type, so there is no other library these can go in.
```
Comment (jonmeow) [inline core/prelude/operators/as.carbon]: Had you considered `core/prelude/types/int.carbon`, similar to the `i32.carbon`?

**Review by jonmeow — COMMENTED** [review]
Generally LG, pursuant to conversation about file size.

**Review by jonmeow — APPROVED** [review]
Approving but not merging. The changes here seem fine to me, and I think my one concern will be addressed through #4534.

**Pull request merged**